pkg/client/query/sharedquerier.go

```diff
@@ -0,0 +1,55 @@
+package query
+
+import (
+	"context"
+
+	"cosmossdk.io/depinject"
+	"github.com/cosmos/gogoproto/grpc"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+var _ client.SharedQueryClient = (*sharedQuerier)(nil)
+
+// sharedQuerier is a wrapper around the sharedtypes.QueryClient that enables the
+// querying of on-chain shared information through a single exposed method
+// which returns an sharedtypes.Session struct
+type sharedQuerier struct {
+	clientConn    grpc.ClientConn
+	sharedQuerier sharedtypes.QueryClient
+}
+
+// NewSharedQuerier returns a new instance of a client.SharedQueryClient by
+// injecting the dependecies provided by the depinject.Config.
+//
+// Required dependencies:
+// - clientCtx
+func NewSharedQuerier(deps depinject.Config) (client.SharedQueryClient, error) {
+	querier := &sharedQuerier{}
+
+	if err := depinject.Inject(
+		deps,
+		&querier.clientConn,
+	); err != nil {
+		return nil, err
+	}
+
+	querier.sharedQuerier = sharedtypes.NewQueryClient(querier.clientConn)
+
+	return querier, nil
+}
+
+// GetParams queries & returns the shared module on-chain parameters.
+//
+// TODO_TECHDEBT(#543): We don't really want to have to query the params for every method call.
+// Once `ModuleParamsClient` is implemented, use its replay observable's `#Last` method
```

Comment:
Depends on how we prioritize it. From our discussion about historical params, I got the impression that `ModuleParamsClient` wouldn't be a high priority for the same reason. It would reduce background network overhead but until we have a reason to believe that that's significantly impacting performance, I don't see another reason to prioritize it.